### PR TITLE
[Improve][Issue-2494] add login type column and hide reset password button when login type is ldap

### LINKED ISSUE #2494: [Feature] improve ldap user management
### Search before asking

- [X] I had searched in the [feature](https://github.com/apache/incubator-streampark/issues?q=is%3Aissue+label%3A%22Feature%22) and found no similar feature requirement.

### Description

Streampark support `Sign in with LDAP` or `Sign in with password`. 

image

Suppose there is a LDAP user {name:user_a，password:123}, when this user login for the first time, Streampark will create a user {name:user_a，password:123}(here we ignore the encryption) in user system. This will lead two security problems:
- user_a are able to `Sign in with password` no matter what the status of LDAP service
- if user_a's password is changed in LDAP, before `Sign in with LDAP` again, user_a can still sign in with the old password.

So I propose to forbid LDAP user `Sign in with password`. Besides, the LDAP user is usually managed by some LDAP client, reset the password of a LDAP user in Streampark is unreasonable, we should hide the reset button for LDAP user in Streampark webui.

### Usage Scenario

_No response_

### Related issues

_No response_

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by zhoulii**
## What changes were proposed in this pull request

Issue Number: close #2494 

## Brief change log

1.add login type column when display user info；
2.hide `reset password` button when login type is ldap；
3.can select login type when create user;

## Verifying this change

- *Manually verified the change by testing locally.* 

## Does this pull request potentially affect one of the following parts
 - Dependencies (does it add or upgrade a dependency): (yes / **no**)

### CODE REVIEW

**Comment by wolfboys** [comment]
cc @wangsizhu0504 PTAL, thx

File: streampark-console/streampark-console-webapp/src/views/system/user/User.vue [hunk 1]
```diff
@@ -104,7 +105,8 @@
             icon: 'bx:reset',
             auth: 'user:reset',
             tooltip: t('system.user.table.reset'),
-            ifShow: () => record.username !== 'admin' || userName.value === 'admin',
+            ifShow: () => (record.username !== 'admin' || userName.value === 'admin')
+              && record.loginType == 'LOCAL',
```
Comment (wolfboys) [inline streampark-console/streampark-console-webapp/src/views/system/user/User.vue]: about 'LOCAL'  Recommend use enum type

File: streampark-console/streampark-console-webapp/src/views/system/user/User.vue [hunk 2]
```diff
@@ -49,6 +49,7 @@
   import { UserListRecord } from '/@/api/system/model/userModel';
   import { useI18n } from '/@/hooks/web/useI18n';
   import Icon from '/@/components/Icon';
+  import {LoginTypeEnum} from "/@/views/base/login/useLogin";
```
Comment (wangsizhu0504) [inline streampark-console/streampark-console-webapp/src/views/system/user/User.vue]: Enumeration is import here but does not seem to be used

File: streampark-console/streampark-console-webapp/src/views/system/user/User.vue (hunk 1 above)
Comment (zhoulii) [reply streampark-console/streampark-console-webapp/src/views/system/user/User.vue]: done

**Review by wangsizhu0504 — APPROVED** [review]
LGTM!

**Pull request merged**